File: scratch/add_locale.dart
import 'dart:io';

void main() {
  final file = File('lib/core/localization/translations.dart');
  final content = file.readAsStringSync();

  final regex = RegExp(r"  'en': \{(.*?)\n  \},", dotAll: true);
  final match = regex.firstMatch(content);
  
  if (match != null) {
    final deBlock = "\n  'de': {" + match.group(1)! + "\n  },";
    final newContent = content.replaceFirst('\n};\n', deBlock + '\n};\n');
    file.writeAsStringSync(newContent);
    print("Added 'de' locale structure to translations.dart");
  } else {
    print("Could not find 'en' block");
  }
}

File: scratch/add_locale.py
import re

with open('lib/core/localization/translations.dart', 'r', encoding='utf-8') as f:
    content = f.read()

en_match = re.search(r"  'en': \{(.*?)\n  \},", content, flags=re.DOTALL)
if en_match:
    de_block = "\n  'de': {" + en_match.group(1) + "\n  },"
    new_content = content.replace('\n};', de_block + '\n};')
    
    with open('lib/core/localization/translations.dart', 'w', encoding='utf-8') as f:
        f.write(new_content)
    print("Added 'de' locale structure to translations.dart")
else:
    print("Could not find 'en' block")
